### PR TITLE
Fix #528

### LINKED ISSUE #528: Lint error makes for a comatose program.
See the program at https://www.khanacademy.org/computer-programming/wat5/5754244333961216 

A syntax error (?) is not flagged by Yoko Oh Noes and the program appears comatose (since it is not run).

### PR DESCRIPTION
**Pull request by EthanLuisMcDonough**
### High-level description of change
Prevents blank slots in arrays from silently crashing the program (Fixes #528).

### Are there performance implications for this change?
No.

### Have you added tests to cover this new/updated code?
Yes. See `output_test.js`.

### Risks involved
None that I know of.

### Are there any dependencies or blockers for merging this?
No.

### How can we verify that this change works?
I added tests.  Opening up the demo and writing something like `var h = [,,]; println(10);` also works.

### CODE REVIEW

File: js/output/shared/ast-walker.js [hunk 1]
```diff
@@ -23,7 +23,10 @@ window.walkAST = function(node, path, visitors) {
             if (Array.isArray(node[prop])) {
                 let i = 0;
                 while (i < node[prop].length) {
-                    let child = node[prop][i];
+                    // Esprima represents empty slots as null, so set child to
+                    // a dummy node if the initial node is falsy in order to prevent
+                    // a silent crash
```
Comment (pamelafox) [inline js/output/shared/ast-walker.js]: Are there other situations where the node would be false-y besides the empty slot? I'm wondering if it's helpful to have a test case for undefined/null/0.

**Comment by pamelafox** [comment]
Oh, and a clarification on the PR template. This question "How can we verify that this change works?" is a question about how to manually verify the change once deployed. Having a manual verification in addition to automated tests helps us verify our deploys and make sure that the code truly made it into the build. You did add a a manual test under this PR's question, so that's what I'll use for the deploy.

File: js/output/shared/ast-walker.js (hunk 1 above)
Comment (EthanLuisMcDonough) [reply js/output/shared/ast-walker.js]: If [modern esprima's type definitions](https://github.com/jquery/esprima/blob/master/src/nodes.ts) are anything to go off of, `null` should be the only falsey value that shows up in node array-properties.  I don't think they'd change their AST structure significantly throughout versions.

Comment (pamelafox) [reply js/output/shared/ast-walker.js]: Okay, thanks, that link clarifies.

**Comment by pamelafox** [comment]
This change was deployed to khanacademy.org yesterday, by the way. Thanks for it!

**Pull request merged**